dockerfiles/nodejs-21.Dockerfile

```
@@ -1,15 +1,17 @@
+# syntax=docker/dockerfile:1.7-labs
 FROM node:21.7-alpine3.19
 
+# Ensures the container is re-built if dependency files change
 ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="package.json,package-lock.json"
 
 WORKDIR /app
 
-COPY package.json ./
-COPY package-lock.json ./
+# .git & README.md are unique per-repository. We ignore them on first copy to prevent cache misses
+COPY --exclude=.git --exclude=README.md . /app
```

Hadolint (2.12.0)

[error] invalid flag: --exclude

(DL1000)